FunNet/Classes/Core/NetworkErrorHandler.swift

```diff
@@ -120,3 +121,192 @@ extension UIAlertController {
+            print("Payload: \(String(data: data ?? Data([]), encoding: .utf8) ?? "None")")
+        }
+    }
+    
+    func errorFunction() -> (NSError?) -> Void {
+        return { err in
+            print("Server Error: \(err?.domain ?? "No title")")
+            print("URL: \(err?.userInfo["url"] ?? "none")")
+            print("Status Code: \(err?.code ?? -1)")
+        }
+    }
+}
+
+protocol VNetworkErrorHandler: NetworkErrorFunctionProvider {
+    var errorMessageMap: [Int:String] { get set }
+    
+    func alert(for error: NSError?) -> UIAlertController
+    func alert(_ title: String, _ message: String) -> UIAlertController
+}
+
+extension VNetworkErrorHandler {
+    func alert(for error: NSError?) -> UIAlertController {
+        if let message = errorMessageMap[error?.code ?? -1] {
+            return alert("\(error?.code ?? -1)", message)
+        } else {
+            return alert("Error \(error?.code ?? -1)", "Description: \(error?.debugDescription ?? "None")\nInfo: \(String(describing: error?.userInfo))")
+        }
+    }
+    
+    func alert(_ title: String, _ message: String) -> UIAlertController {
+        let alert = UIAlertController(title: title, message: message, preferredStyle: UIAlertController.Style.alert)
+        alert.addAction(UIAlertAction(title: "OK", style: .cancel, handler: { (action) in
+            alert.dismiss(animated: true, completion: nil)
+        }))
+        return alert
+    }
+}
+
+public class DebugNetworkErrorHandler: VNetworkErrorHandler {
+    
+    
+    open var errorMessageMap: [Int:String] = [
+        400 : "Bad request",
+        401 : "Unauthorized",
+        402 : "Payment required",
+        403 : "Forbidden",
+        404 : "Not found",
+        405 : "HTTP Method not allowed",
+        406 : "Content type in Accept header is unavailable",
+        408 : "Request timed out",
+        409 : "Conflict in requested resource",
+        410 : "Resource is permanently unavailable",
+        411 : "Length required",
+        412 : "Precondition failed",
+        413 : "Payload too large",
+        414 : "URI too long",
+        415 : "Unsupported mediatype",
+        416 : "Range not satisfiable",
+        417 : "Expectation failed",
+        418 : "This server is, in fact, a teapot",
+        420 : "420 error, dank dude",
+        421 : "Unauthorized",
+        422 : "Unable to process payload",
+        429 : "Too many requests",
+        431 : "Headers too large",
+        451 : "Unavailable for legal reasons",
+        500 : "Internal server error",
+        501 : "This functionality is not implemented",
+        502 : "Bad internet gateway",
+        503 : "Server currently unavailable",
+        505 : "HTTP version not supported",
+        511 : "Network authentication required"
+    ].merging(urlLoadingErrorCodesDict, uniquingKeysWith: { _, key in return key })
+    
+    func errorFunction() -> (NSError?) -> Void {
+        return { err in
+            self.alert(for: err).show()
```

Comment:
Ok, alas, `show` doesn't work anymore. I've only left it in for the legacy apps that use it, but it doesn't work post iOS... 12? Or 13? I think? What I'd like to be able to do is use these error handlers to bind networkCalls to VCs, a la:
```swift
func verboseErrors(displayedIn vc: UIViewController) -> (ReactiveNetworkResponder) -> Void {
    let display = vc.presentClosure()
    return { responder in
        responder.serverErrorSignal.observeValues(VerboseNetworkErrorHandler().alert(for:) >>> display)
        responder.errorSignal.observeValues(VerboseURLLoadingErrorHandler().alert(for:) >>> display)
    }
}
```